### DISCUSSION TITLE
Select Multiple Values

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/itwinjs-core#3282] [Q&A]
I have a select dropdown menu and I want to be able to choose multiple values and not just a single value. I am thinking doing it with a checkboxes in dropdown. Kindly guide me on how I can accomplish it.

**Comment by ayushibasaiwal** [comment]
Hello, I have tried to get the required functionality of having a select dropdown having multiple checkboxes. I have used HTML, CSS and JavaScript for this.

create a HTML file:
[index.txt](https://github.com/iTwin/itwinjs-core/files/9037699/index.txt)

create a CSS file:
.multiselect {
    width: 200px;
  }
  
  .selectBox {
    position: relative;
  }
  
  .selectBox select {
    width: 100%;
    font-weight: bold;
  }
  
  .overSelect {
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
  }
  
  #checkboxes {
    display: none;
    border: 1px #dadada solid;
  }
  
  #checkboxes label {
    display: block;
  }
  
  #checkboxes label:hover {
    background-color: #1e90ff;
  }
  
create a JS file:
var expanded = false;
function toggleCheckboxesVisibility() {
  var checkboxes = document.getElementById("checkboxes");
  if (!expanded) {
    checkboxes.style.display = "block";
    expanded = true;
  } else {
    checkboxes.style.display = "none";
    expanded = false;
  }
}

Explanation:
To add the functionality we specify a JavaScript function that is called when the user clicks on the div that contains our select element 
(div-> class="selectBox" onclick="toggleCheckboxesVisibility()")
We also create div that contains our checkboxes and style it using CSS. 
The toggleCheckboxesVisibility JavaScript function just changes <div id="checkboxes"> value of CSS display property from "none" to "block" and vice versa.
The display of div of id 'checkboxes' is none and when user click on select dropdown it changes to display "block". 
We can get checked value (if the checkbox is checked or not) by :

let checkbox1= document.getElementById("one");
checkbox1.onclick=(e)=>{
  console.log(e.target.checked);
}

**Comment by veekeys** [comment]
Hello @hanan-commits-work,

You might be interested in this feature https://github.com/iTwin/iTwinUI-react/issues/344
Development is ongoing, so this might be a part of iTwinUI `Select` component pretty soon.
